src/serializers/computed_fields.rs

```diff
@@ -190,3 +194,60 @@ impl<'py> Serialize for ComputedFieldSerializer<'py> {
         s.serialize(serializer)
     }
 }
+
+fn has_ser_function(schema: &PyDict) -> bool {
+    let py = schema.py();
+    let ser_schema = schema
+        .get_as::<&PyDict>(intern!(py, "serialization"))
+        .unwrap_or_default();
+    ser_schema.is_some_and(|s| s.contains(intern!(py, "function")).unwrap_or_default())
+}
+
+fn get_next_value<'a>(
+    field: &'a ComputedField,
+    input_value: &'a PyAny,
+    ob_type_lookup: &'a ObTypeLookup,
+) -> PyResult<&'a PyAny> {
+    let py = input_value.py();
+    // Backwards compatiability.
+    let mut legacy_attr_error: Option<PyErr> = None;
+    let legacy_result = match ob_type_lookup.get_type(input_value) {
```

Comment:
I don't think it's fair to call this "backwards compatibility" when this is still expected to be the main code path for models and dataclasses. 

I wonder if there might be a more unified way. For a model or dataclass `A` with computed field `b`, the analogous functionality really seems to be `A.b.__get__(instance)`. For a `TypedDict` it looks like that also works:

```python
>>> class Bar(TypedDict):
...      @property
...      def y(self):
...          return 434
...
>>> Bar.y.__get__({})
434
```

So maybe what we really want, in all cases, is

```rust
let property_value = input_value.get_type().getattr(field.property_name_py.as_ref(py))?.call_method1("__get__", (input_value,))?;
```

Reply:
My thinking was that you provide a serialization function to a computed field, doesn't matter if the type is a `Model`, `TypedDict`, `Dataclass`, etc.

The default behavior is to compute the computed value from the function provided in the serialization schema and then it gets set in the `output_dict`: https://github.com/pydantic/pydantic-core/blob/866eb2da06321cf4c99118b2331bf7eff7e81e1c/src/serializers/computed_fields.rs#L154

This seems more generalizable to all computed fields instead of relying on the computed field defined as an attribute on the input value.

However, I might be missing some context on how computed fields are used by https://github.com/pydantic/pydantic.

Reply:
I'm not following all that closely but my 2c is that ideally we extract the function from the thing in `pydantic` and not in `pydantic-core` so that:
1. We have more flexibility. It's easier to hack things (like rebuild `__mro__` based on `__orig_bases__` which we do for TypedDict)
2. It ensures that we do this at schema build time and not runtime